app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,39 @@
+module SessionsHelper
+  def log_in(user)
+    session[:user_id] = user.id
+  end
+
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def log_off
+    forget(current_user)
+    session.delete(:user_id)
+    @current_user = nil
+  end
+
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
+      user = User.find_by(id: user_id)
+      if user&.authenticated?(cookies[:remember_token])
+        log_in user
+        @current_user = user
+      end
+    end
```

Comment:
Too big !!!
By only using `cookies`, you could shrink that in one line

Reply:
sorry I can not see how that could be